cdisc_rules_engine/services/datasetjson_metadata_reader.py

```diff
@@ -131,9 +131,10 @@ def _convert_variable_types(self):
         rule authors use.
         """
         rule_author_type_map: dict = {
-            "string": "Char",
-            "integer": "Num",
+            "decimal": "Num",
             "float": "Num",
+            "integer": "Num",
+            "string": "Char",
         }
```

Comment:
The Dataset-JSON specification describes `type` as:
> Type of the variable. Allowed values: "string", "integer", "decimal", "float", "double", "boolean". See ODM types for details.

If a mapping for "decimal" is being added, wouldn't it also make sense to add mappings for "double" and "boolean"?

Reply:
Not sure what to map boolean to because I don't think it would exist in SDTM. I mapped it to num because this is what the sas json engine does according to Lex

Reply:
I had the same question. I'll try to remember to bring it up during a future dataset JSON meeting. I've already seen other people map it to CHAR with Yes/No instead. 😬